src/conversion.rs

```diff
@@ -336,30 +296,30 @@ where
     }
 }
 
-impl<T> FromPyObject<'_> for T
+impl<T> FromPyObject<'_, '_> for T
 where
     T: PyClass + Clone,
 {
-    fn extract_bound(obj: &Bound<'_, PyAny>) -> PyResult<Self> {
+    fn extract(obj: Borrowed<'_, '_, PyAny>) -> PyResult<Self> {
         let bound = obj.downcast::<Self>()?;
         Ok(bound.try_borrow()?.clone())
     }
 }
 
-impl<'py, T> FromPyObject<'py> for PyRef<'py, T>
+impl<'py, T> FromPyObject<'_, 'py> for PyRef<'py, T>
 where
     T: PyClass,
 {
-    fn extract_bound(obj: &Bound<'py, PyAny>) -> PyResult<Self> {
+    fn extract(obj: Borrowed<'_, 'py, PyAny>) -> PyResult<Self> {
         obj.downcast::<T>()?.try_borrow().map_err(Into::into)
     }
 }
 
-impl<'py, T> FromPyObject<'py> for PyRefMut<'py, T>
+impl<'py, T> FromPyObject<'_, 'py> for PyRefMut<'py, T>
```

Comment:
And then the follow-up future PR will be what to do about `PyRef`. Should we split into `PyRef` / `PyRefBorrowed`, or keep the one type `PyRef<'a, 'py, T>` and accept some breakage? 🤔

Reply:
Hmm, I think I we want to provide both variants, I would prefer `PyRef<'a, 'py, T>` and `PyRefOwned<'py, T>` since that would be more consistent with `FromPyObjectOwned`. 

In any case we would need to find a good way to differentiate their constructors. Neither `obj.borrow_borrowed()` nor `obj.borrow_owned()` feel to appealing to me 😄

Reply:
Agreed, it's tough to find names. I'm somewhat tempted to wait until after 0.23 and to first see how all these trait changes we've already made play out before we change `PyRef`.